src/engraving/libmscore/ambitus.h

```diff
@@ -126,6 +115,38 @@ class Ambitus final : public EngravingItem
     EngravingItem* nextSegmentElement() override;
     EngravingItem* prevSegmentElement() override;
 
+    static AccidentalType accidentalType(int tpc, Key key);
+    static int staffLine(int tpc, int pitch, ClefType clf);
```

Comment:
I don't see yet where these methods come from and where they are used...

Reply:
There was code duplication in layout method, I removed duplications, and also decided that this logic deserves to be a separate method, that is, it has independent value.
https://github.com/musescore/MuseScore/blob/master/src/engraving/layout/dev/tlayout.cpp#L537
https://github.com/musescore/MuseScore/blob/master/src/engraving/layout/dev/tlayout.cpp#L533

These methods are also used in the layout
https://github.com/musescore/MuseScore/pull/18801/files#diff-49b7f7bbf83fa3a3e79cf25cb143809bdc2bbf93146782320bf025a6227b99cfR491
https://github.com/musescore/MuseScore/pull/18801/files#diff-49b7f7bbf83fa3a3e79cf25cb143809bdc2bbf93146782320bf025a6227b99cfR494